src/lib/gameService.ts

```diff
@@ -12,41 +13,172 @@ export class GameService {
 			code: this.generateCode(),
 			currentPlayer: undefined,
 			players: [],
-			currentBid: undefined
+			currentBid: undefined,
+			maxPlayers: 8,
+			numberOfPlayers: 0,
+			initialDiceCount: 6
 		};
 
 		this.repository.saveGame(game);
 
 		return game.code;
 	}
 
-	public async addPlayer(name: string, gameCode: string): Promise<string> {
-		throw new Error('not implemented');
+	public async addPlayer(playerName: string, gameCode: string): Promise<string> {
+		const game: Game | undefined = await gameRepository.getGame(gameCode);
+
+		if (!game) {
+			throw new Error('Game is undefined');
+		}
+
+		if (game.state === GameState.InProgress) {
+			throw new Error('The game is already in progress');
+		} else if (game.state === GameState.Finished) {
+			throw new Error('The game has already finished');
+		}
+
+		const newPlayer: Player = {
+			name: playerName,
+			code: gameCode,
+			dice: []
+		};
+
+		if (game.players.length <= game.maxPlayers) {
+			game.players.push(newPlayer);
+		} else {
+			throw new Error('max player cap check failed');
+		}
+
+		this.repository.saveGame(game);
+
+		return playerName;
+
+		// throw new Error('not implemented');
 	}
 
-	public async getPlayers(name: string, gameCode: string): Promise<PlayerDto[]> {
-		throw new Error('not implemented');
+	// Should this instead return OpponentDto[]?
```

Comment:
My intention for this function was just to return player names for a lobby, it can be really basic for now